wire/msgtx.go

```diff
@@ -968,25 +968,66 @@ func NewMsgTx(version int32) *MsgTx {
 }
 
 // readOutPoint reads the next sequence of bytes from r as an OutPoint.
+//
+// DEPRECATED: Use readOutPointBuf instead.
 func readOutPoint(r io.Reader, pver uint32, version int32, op *OutPoint) error {
+	buf := binarySerializer.Borrow()
+	err := readOutPointBuf(r, pver, version, op, buf)
+	binarySerializer.Return(buf)
+	return err
+}
+
+// readOutPointBuf reads the next sequence of bytes from r as an OutPoint.
+//
+// If b is non-nil, the provided buffer will be used for serializing small
+// values.  Otherwise a buffer will be drawn from the binarySerializer's pool
```

Comment:
nit: this part of the comment should be on `readOutPoint` I think, since we don't draw a buffer from the binarySerializer here. Same below for `writeOutPointBuf`.